Feature/team24 333

Mohamad11Dab opened the pull request:
**If need help setting up project:**
1. `./run_project_docker.sh` at prism root.
2.  Click Y for downloading mongodump data and if you need docker images built, click Y for that option too.


**Setup:**
Please make sure you have client,server and db running.
Run `docker ps` to see your available containers. If no containers exist, run `docker compose up` or `docker compose up --build`
```bash
git checkout feature/TEAM24-333
```

**Testing:**
`yarn cypress:run:local --spec cypress/e2e/EditNodePositionFormTest.cy.ts `
see how one of the nodes' name in prism_uat databse will change to "Theater"

Ticket:
https://elipse-uq.atlassian.net/browse/TEAM24-333?atlOrigin=eyJpIjoiMmJiNmI3NWNkNTY0NDA3ZmJhMWRjN2RjODA1ODViZDYiLCJwIjoiaiJ9

Mohamad11Dab committed:
Admin can now change the node name

Mohamad11Dab committed:
The server side of it

Mohamad11Dab committed:
Merge remote-tracking branch 'origin/develop' into feature/TEAM24-333

Mohamad11Dab committed:
removed .only

pipw commented:
* I don't think that a full on docker build should be required for this code-review
* UI functionality would be better tested in UAT once merged into dev
* `yarn outdated` should be run and packages updated where possible or commented where not (ie new ticket?)
* `yarn test` should pass

Mohamad11Dab committed:
fixed test cases

Mohamad11Dab committed:
Updated packages

Mohamad11Dab commented:
Updated some of the packages. The below packages are causing the app to fail so they should be handled and updated in a separate ticket 

![Screenshot from 2024-04-23 16-05-08](https://github.com/UQ-eLIPSE/prism/assets/114811082/f90033ca-bb95-4370-915e-2a43053db1c5)

Mohamad11Dab committed:
fixed error cases in client side jest tests
